File: #0409.longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {

        map<char, int> m;
        for (char c : s) {
            if (m.find(c) == m.end()) {
                m[c] = 1;
            } else {
                m[c]++;
            }
        }
        
        int result = 0;
        for (map<char, int>::iterator it = m.begin(); it != m.end(); it++) {
            result += it->second / 2 * 2;
            if (result % 2 == 0 && it->second % 2 == 1) {
                result++;
            }
        }
        return result;
    }
};

File: #0409.longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    let result = 0;
    const set = new Set();
    for (let i = 0; i<s.length; i++) {
        if (set.has(s[i])) {
            result += 2;
            set.delete(s[i]);
        } else { 
            set.add(s[i]);
        }
    };
    return result + (set.size > 0 ? 1 : 0);
};
